packages/sources/onchain-gas/src/adapter.ts

```diff
@@ -0,0 +1,69 @@
+import { Builder, Requester, Validator } from '@chainlink/ea-bootstrap'
+import {
+  Config,
+  ExecuteWithConfig,
+  ExecuteFactory,
+  AdapterRequest,
+  APIEndpoint,
+  MakeWSHandler,
+} from '@chainlink/types'
+import { makeConfig } from './config'
+import * as endpoints from './endpoint'
+
+export const execute: ExecuteWithConfig<Config> = async (request, context, config) => {
+  return Builder.buildSelector(request, context, config, endpoints)
+}
+
+export const endpointSelector = (request: AdapterRequest): APIEndpoint =>
+  Builder.selectEndpoint(request, makeConfig(), endpoints)
+
+export const makeExecute: ExecuteFactory<Config> = (config) => {
+  return async (request, context) => execute(request, context, config || makeConfig())
+}
+
+export const makeWSHandler = (config?: Config): MakeWSHandler => {
+  return () => {
+    const defaultConfig = config || makeConfig()
+    return {
+      connection: {
+        url: defaultConfig.api.baseURL,
+      },
+      toSaveFromFirstMessage: (message: any) => ({
+        subscriptionId: message.params.subscription,
+      }),
+      noHttp: true,
+      subscribe: (input) => ({
+        id: input.id,
+        method: 'eth_subscribe',
+        params: ['newHeads'],
+      }),
+      unsubscribe: (input, subscriptionParams) => ({
+        id: input.id,
+        method: 'eth_unsubscribe',
+        params: [subscriptionParams.subscriptionId],
+      }),
+      subsFromMessage: (_, subscriptionMsg) => {
+        return {
+          id: subscriptionMsg.id,
+          method: 'eth_subscribe',
+          params: ['newHeads'],
+        }
+      },
+      isError: () => false,
+      filter: (message) => message.method === 'eth_subscription',
+      toResponse: async (message: any, input: AdapterRequest) => {
+        const validator = new Validator(input, endpoints.gas.inputParameters)
+        if (validator.error) throw validator.error
+        const hexedBlockNum: string = message.params.result.number
+        const medianGasPrices = await endpoints.gas.getTransactionsInPastBlocks(
```

Comment:
One thing that we might want to account for is I know the node used to have an issue when the result contained an array. So we might want the response data to look like:
```json
{
    "data": {
        "values": [101768070829],
        "result": 101768070829
    },
    "result": 101768070829
}
```

This also means when there are multiple blocks queried, we'll want another param to specify which individual value we come to for the `result`.